Add new change log entry for deprecating shared use of old accounting datatypes (#184)
codatio/codat-docs: blog/230710-deprecation-bankaccount-banktransaction.md

---
title: "Upcoming 2023-07-10: Deprecation of the use of bankAccounts and bankTransactions data types for banking data connections"
date: "2023-07-10"
tags: ["Deprecation"]
draft: true
authors: GraceCodat
---

> On July 10, 2023, Codat will deprecate the use of `bankAccounts` and `bankTransactions` data types for syncing banking data connections via Plaid, TrueLayer and Basiq. The use of these data types for accounting data connections will remain unchanged.

Reviewer: ```suggestion
On July 10, 2023, Codat will deprecate the use of the `bankAccount` and `bankTransactions` data types for syncing banking data connections via Plaid, TrueLayer, and Basiq. The use of these data types for accounting data connections will remain unchanged.
```

Author: `bankAccounts` should have the 's'

Requests made to the following endpoints for banking data connections will no longer trigger a sync:

- `POST /companies/{companyId}/data/queue/{bankAccounts}`
- `POST /companies/{companyId}/data/queue/{bankTransactions}`

> After the change is implemented a call to either of these endpoints for a Plaid, TrueLayer or Basiq data connection ID will return a 400 response. 

Reviewer: ```suggestion
After the change is implemented, a call to either of these endpoints for a Plaid, TrueLayer, or Basiq data connection ID will return a 400 response. 
```

## Action required​

> Switch to using the banking data types: [`banking-accounts`](https://docs.codat.io/banking-api#/schemas/Account) and [`banking-transactions`](https://docs.codat.io/banking-api#/schemas/Transaction). Enable these in your settings in the Portal.

Reviewer: ```suggestion
If you haven't already done so, switch to using the new banking data types: [`banking-accounts`](https://docs.codat.io/banking-api#/schemas/Account) and [`banking-transactions`](https://docs.codat.io/banking-api#/schemas/Transaction). Enable these in your settings in the Portal.
```

Author: I dont think we need to include the word 'new'. The banking datatypes have been available for a long time, we just haven't requested people use them. I also want to avoid confusion for clients who are using the banking datatypes but read this.. They may worry that we have 'new' banking ones when we dont.

Reviewer: Noted. I will delete the suggestion.

You will also need to update any parts of your code that expect the response body from the `banking-accounts` and `banking-transactions` endpoints. 
For examples of the new response bodies for these data types, refer to the API reference.

## Expected impact if no action is taken​

> After July 10, 2023, calls made using `bankAccounts` and `bankTransactions` data types for a Plaid, TrueLayer or Basiq data connection ID will return a 400 error:

Reviewer: ```suggestion
After July 10, 2023, any calls made using the `bankAccount` and `bankTransactions` data types for a Plaid, TrueLayer, or Basiq data connection ID will return a 400 error. For example:
```

---
title: "Upcoming 2023-07-10: Deprecation of the use of bankAccount and bankTransaction data types for banking data connections"
date: "2023-07-10"
tags: ["Deprecation"]
draft: true

> authors: d-laing

Reviewer 2: ```suggestion
authors: GraceCodat
```
